### ISSUE TITLE
ArrayIndexOutOfBoundsException when loading unit

### ISSUE THREAD

**Issue by HammerGS** [MegaMek/megameklab#1227]
Reporting from Discord. 

### Environment
What version of MekLab does your issue apply to?  MegaMek v0.49.12-SNAPSHOT
What operating system are you using?  Windows
What java version are you using?  Java 11

### Description
```
11:33:11,478 ERROR [megamek.common.loaders.MtfFile] {Mech Cache Loader}
megamek.common.loaders.MtfFile.getEntity(MtfFile.java:516) - 
java.lang.ArrayIndexOutOfBoundsException: Index 12 out of bounds for length 12
	at megamek.common.Entity.addCritical(Entity.java:4621)
	at megamek.common.loaders.MtfFile.parseCrits(MtfFile.java:811)
	at megamek.common.loaders.MtfFile.getEntity(MtfFile.java:451)
	at megamek.common.MechFileParser.parse(MechFileParser.java:153)
	at megamek.common.MechFileParser.<init>(MechFileParser.java:44)
	at megamek.common.MechFileParser.<init>(MechFileParser.java:37)
	at megamek.common.MechSummaryCache.loadMechsFromDirectory(MechSummaryCache.java:621)
	at megamek.common.MechSummaryCache.loadMechsFromDirectory(MechSummaryCache.java:580)
	at megamek.common.MechSummaryCache.checkForChanges(MechSummaryCache.java:253)
	at megamek.common.MechSummaryCache.loadMechData(MechSummaryCache.java:242)
	at megamek.common.MechSummaryCache.lambda$getInstance$0(MechSummaryCache.java:82)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

### Files
[Sylverant_Mobile_Headquarters.zip](https://github.com/MegaMek/megameklab/files/10460093/Sylverant_Mobile_Headquarters.zip)

**Comment by HammerGS** [comment]
```
12:26:02,239 ERROR [megamek.common.loaders.MtfFile] {AWT-EventQueue-0}
megamek.common.loaders.MtfFile.getEntity(MtfFile.java:516) - 
java.lang.ArrayIndexOutOfBoundsException: Index 12 out of bounds for length 12
	at megamek.common.Entity.addCritical(Entity.java:4627)
	at megamek.common.loaders.MtfFile.parseCrits(MtfFile.java:811)
	at megamek.common.loaders.MtfFile.getEntity(MtfFile.java:451)
	at megamek.common.MechFileParser.parse(MechFileParser.java:153)
	at megamek.common.MechFileParser.<init>(MechFileParser.java:44)
	at megamek.common.MechFileParser.<init>(MechFileParser.java:37)
	at megameklab.ui.MenuBar.loadUnitFromFile(MenuBar.java:1460)
	at megameklab.ui.MenuBar.lambda$createLoadMenu$15(MenuBar.java:329)
```

and

```
megamek.common.MechFileParser.<init>(MechFileParser.java:46) - 
java.lang.Exception: java.lang.ArrayIndexOutOfBoundsException: Index 12 out of bounds for length 12
	at megamek.common.loaders.MtfFile.getEntity(MtfFile.java:517)
	at megamek.common.MechFileParser.parse(MechFileParser.java:153)
	at megamek.common.MechFileParser.<init>(MechFileParser.java:44)
	at megamek.common.MechFileParser.<init>(MechFileParser.java:37)
	at megameklab.ui.MenuBar.loadUnitFromFile(MenuBar.java:1460)
	at megameklab.ui.MenuBar.lambda$createLoadMenu$15(MenuBar.java:329)
```

**Comment by Amaroq-Clearwater** [comment]
Got another one.
[Superheavy Tripod Omni.zip](https://github.com/MegaMek/megameklab/files/10473005/Superheavy.Tripod.Omni.zip)

**Comment by HammerGS** [comment]
And you will continue to get them till we fix it.

**Comment by Amaroq-Clearwater** [comment]
I know. But I'm making a note because this one has nothing on it.

**Comment by Sleet01** [comment]
Did some investigation and it looks like the root cause is that Superheavy components get their slots allocated correctly (halved if >1.0) but that each entry in the slot records its original SIZE: value; if e.g. a 4.0-ton Communications Equipment component is put into a 2-slot space on a super-heavy with no extra space, it gets saved as _two_ slots - _each_ with a SIZE: 4.0.

I think this causes the component to be expanded to take up 4 slots at load time, which is incorrect - at least, for a Superheavy.  If I manually edit the SIZE: value to 2.0 but leave the number of consumed slots the same, a file that exhibits this bug will then load.  Of course, in this case pulling the component off the Criticals diagram sets it to a 2.0t/1 crit item (because this is determined by the SIZE: value and the Superheavy modifier).

I'd propose adding a load-time special handler for Superheavies only, that uses the stored crit _slots_ of a component with a SIZE value, but otherwise sets its mass and SIZE value the same.

**Comment by HammerGS** [comment]
A permanent fix is preferable to a work around

**Comment by Sleet01** [comment]
@HammerGS I have a proposed fix referencing this issue, but in fact all changes are in MegaMek code.  Should I do the PR here, or how does that work?

**Comment by HammerGS** [comment]
You found PR it of the code base (MegaMek) and reference the MML issue number in the notes.

**Comment by NickAragua** [comment]
Fixed by MM PR #4404
